### PR TITLE
feat(dynamic-sampling): Add support for `trace.replay_id` in matching rules

### LINKED ISSUE #47769: Replay Bias
Ensure that transactions with a Replay ID on the DSC are sampled so that the trace view for any given Replay is complete.
- Jernej asked us to give heads-up in #discuss-ingest once we start working on this and give a little context to the team

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR adds the ability to read the `replay_id` field in the `DynamicSamplingContext`.

Closes: https://github.com/getsentry/sentry/issues/47769

### CODE REVIEW

File: relay-sampling/src/lib.rs [hunk 1]
```diff
@@ -316,6 +316,27 @@ impl NotCondition {
     }
 }
 
+/// Has condition combinator.
+///
+/// Creates a condition that is true when a specific field is present.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct HasCondition {
+    field: String,
+}
```
Comment (jjbayer) [inline relay-sampling/src/lib.rs]: AFAIK This should also work with a composite `{"not": {"eq": null}}` condition. The reason why I would prefer that is that it is already supported by external relays.

**Review by jjbayer — CHANGES_REQUESTED** [review]

File: relay-sampling/src/lib.rs (hunk 1 above)
Comment (iambriccardo) [reply relay-sampling/src/lib.rs]: Semantically it is equivalent yes, I can for sure change that. Thanks for the heads up!

File: relay-sampling/src/lib.rs [hunk 2]
```diff
@@ -856,6 +856,10 @@ impl FieldValueProvider for DynamicSamplingContext {
                 None => Value::Null,
                 Some(ref s) => s.as_str().into(),
             },
+            "trace.replay_id" => match self.replay_id {
+                None => Value::Null,
+                Some(ref s) => Value::String(s.to_string()),
```
Comment (jjbayer) [inline relay-sampling/src/lib.rs:861]: nit: For consistency with the code above

```suggestion
                Some(ref s) => s.as_str().into(),
```

**Review by jjbayer — APPROVED** [review]

File: relay-sampling/src/lib.rs (hunk 2 above)
Comment (iambriccardo) [reply relay-sampling/src/lib.rs:861]: I have tried that but `Uuid` has no method `as_str()`.

**Pull request merged**